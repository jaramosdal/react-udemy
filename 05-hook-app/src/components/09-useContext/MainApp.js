import React from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {
    const user = {
        id: 1234,
        name: 'Javi',
        email: 'javi@email.com'
    }

    return (
        <UserContext.Provider value={ user }>
            <AppRouter />
        </UserContext.Provider>
    )
}
 