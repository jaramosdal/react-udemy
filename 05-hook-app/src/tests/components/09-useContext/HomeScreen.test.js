import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen />', () => {

    const user = {
        name: 'Javier',
        email: 'javi@email.com'
    }
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen  />
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

});