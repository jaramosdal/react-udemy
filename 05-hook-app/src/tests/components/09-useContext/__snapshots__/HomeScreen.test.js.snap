// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Javier",
   "email": "javi@email.com"
}
  </pre>
</div>
`;
